src/main/groovy/io/jenkins/infra/repository_permissions_updater/ArtifactoryPermissionsUpdater.groovy

```diff
@@ -121,6 +122,25 @@ class ArtifactoryPermissionsUpdater {
                 }
             }
 
+            String artifactId = definition.name
+            for (String path : definition.paths) {
+                if (path.substring(path.lastIndexOf('/') + 1) != artifactId) {
+                    // We could throw an exception here, but we actively abuse this for unusually structured components
+                    LOGGER.log(Level.WARNING, "Unexpected path: " + path + " for artifact ID: " + artifactId)
+                }
+                String groupId = path.substring(0, path.lastIndexOf('/')).replace('/', '.')
+
+                String key = groupId + ":" + artifactId
+
+                if (maintainersByComponent.containsKey(key)) {
+                    LOGGER.log(Level.WARNING, "Duplicate maintainers entry for component: " + key)
+                }
+                // A potential issue with this implementation is that groupId changes will result in lack of maintainer information for the old groupId.
+                // In practice this will probably not be a problem when path changes here and subsequent release are close enough in time.
+                // Alternatively, always keep the old groupId around for a while.
+                maintainersByComponent.computeIfAbsent(key, { _ -> new ArrayList<>(Arrays.asList(definition.developers)) })
```

Comment:
thoughts on pulling out display names for plugin site? we're changing from:

> Tim Jacomb

to:
> timja

with this change

Reply:
If we apply this directly it would be worse, yes. My current plan is to add a new https://github.com/jenkins-infra/infra-reports/ that provides a mapping from user ID to user name (perhaps from Jira, to not require LDAP access) for maintainer accounts. TBD whether that'll make it into the first iteration, or will be a future enhancement. [INFRA-2926](https://issues.jenkins.io/browse/INFRA-2926) is a bit annoying in this context though.

Reply:
@timja @oleg-nenashev https://github.com/jenkins-infra/infra-reports/pull/27